src/Navigation/Drawer/DrawerStyle.ts

```diff
@@ -44,38 +44,47 @@ const closeToLeft = keyframes`
 const fadeIn = keyframes`
   from {
     opacity: 0;
+    visibility: hidden;
+    background-color: transparent;
   }
-
+  
   to {
     opacity: 1;
+    visibility: visible;
+    background-color: rgba(0, 0, 0, 0.5);
   }
-`;
+  `;
 
 const fadeOut = keyframes`
   from {
     opacity: 1;
+    visibility: visible;
+    background-color: rgba(0, 0, 0, 0.5);
   }
-
+  
   to {
     opacity: 0;
+    visibility: hidden;
+    background-color: transparent;
   }
 `;
 
 export const DrawerContainer = styled.div<DrawerContainerProps>`
   position: fixed;
-  display: ${({ isDisplay }) => (isDisplay ? 'block' : 'none')};
   min-height: 100vh;
   max-height: 100vh;
   width: 100vw;
   top: 0px;
   left: 0px;
-  background-color: rgba(0, 0, 0, 0.5);
+  opacity: ${({ open }) => (open ? '1' : '0')};
+  visibility: ${({ open }) => (open ? 'visible' : 'hidden')};
+  background-color: ${({ open }) =>
+    open ? 'rgba(0, 0, 0, 0.5)' : 'transparent'};
   animation: ${({ open }) => (open ? fadeIn : fadeOut)} 0.3s ease-in-out;
-  z-index: ${({ isDisplay }) => (isDisplay ? '9999' : '-1')};
+  z-index: 9999;
```

Comment:
z-index should be this only when it's opened, right? Should we have the condition?

Reply:
Somehow it seemed to make a difference when I set it to `-1`, it was glitching out. I think having it at this index at all times is fine.

Reply:
Oh even if we set index when it's open and do not set any index value when it's not open?

Reply:
Let me try that...

Reply:
Yeah not setting it doesn't work either. The problem is, it needs to be at z-index 9999 when the animation is running. Otherwise the overlay could get rendered behind other elements.